tests/test_trailedEdgeSources.py

```diff
@@ -0,0 +1,220 @@
+#
+# This file is part of meas_extensions_trailedSources.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import numpy as np
+import unittest
+import lsst.utils.tests
+import lsst.meas.extensions.trailedSources
+from lsst.meas.base.tests import AlgorithmTestCase
+from lsst.utils.tests import classParameters
+
+# Trailed-source length, angle, and centroid, one on the edge, one not.
+Ls = np.array([5, 5])
+thetas = np.array([100, 0])
+xcs = np.array([100, 20])
+ycs = np.array([100, 20])
+
+
+class TrailedEdgeSource:
+    """Holds a set of true trail parameters.
+    """
+
+    def __init__(self, instFlux, length, angle, xc, yc):
+        self.instFlux = instFlux
+        self.length = length
+        self.angle = angle
+        self.center = lsst.geom.Point2D(xc, yc)
+        self.x0 = xc - length / 2 * np.cos(angle)
+        self.y0 = yc - length / 2 * np.sin(angle)
+        self.x1 = xc + length / 2 * np.cos(angle)
+        self.y1 = yc + length / 2 * np.sin(angle)
+
+
+# "Extend" meas.base.tests.TestDataset
+class TrailedTestDataset(lsst.meas.base.tests.TestDataset):
+    """A dataset for testing trailed source measurements.
+    Given a `TrailedSource`, construct a record of the true values and an
+    Exposure.
+    """
+
+    def __init__(self, bbox, threshold=10.0, exposure=None, **kwds):
+
+        super().__init__(bbox, threshold, exposure, **kwds)
+
+    def addTrailedSource(self, trail):
```

Comment:
This is a clever way of generating fake trails! Do you have any images showing that it is doing what you think it is? I worry it is easy to have a geometry error or be off by a factor of 2 or something "silly" without actually seeing any fake trails and their positions.